src/mount/mount-tool.c

```diff
@@ -296,6 +296,31 @@ static int parse_argv(int argc, char *argv[]) {
                         log_error("Listing devices only supported locally.");
                         return -EOPNOTSUPP;
                 }
+        } else if (arg_action == ACTION_UMOUNT) {
+                if (optind >= argc) {
+                        log_error("At least one argument required.");
+                        return -EINVAL;
+                }
+
+                if (argc > optind+1) {
+                        log_error("At most one argument required.");
+                        return -EINVAL;
+                }
+
+                if (is_device_path(argv[optind])) {
```

Comment:
OK. But why not move lines 310-322 into `stop_mounts` (adjusting the arguments as necessary)?